src/index/base.cpp

```diff
@@ -149,7 +149,7 @@ void BaseIndex::ThreadSync()
                     Commit();
                     break;
                 }
-                if (pindex_next->pprev != pindex && !Rewind(pindex, pindex_next->pprev)) {
+                if (pindex_next->pprev != pindex && !Rewind(pindex, pindex_next->pprev, consensus_params)) {
```

Comment:
why f4013af8636636c6bb30b43d913283e9dcee0134?

Reply:
Because `GetBlockSubsidy` needs it (see [`Rewind()`](https://github.com/dashpay/dash/blob/5ccb4a324f4c348b6b738a78b44b976794d33f92/src/index/coinstatsindex.cpp#L263) -> [`ReverseBlock()`](https://github.com/dashpay/dash/blob/5ccb4a324f4c348b6b738a78b44b976794d33f92/src/index/coinstatsindex.cpp#L292) -> [`GetBlockSubsidy()`](https://github.com/dashpay/dash/blob/5ccb4a324f4c348b6b738a78b44b976794d33f92/src/index/coinstatsindex.cpp#L405) -> calculations related to halving and superblocks ([source](https://github.com/dashpay/dash/blob/5ccb4a324f4c348b6b738a78b44b976794d33f92/src/validation.cpp#L1150-L1163)).

Earlier iterations of this PR had me attempting to avoid passing `consensus_params` by splitting up the base subsidy calculation into its own function (see [here](https://github.com/dashpay/dash/commit/ed3c05cba05c9acbf5758eb7360230163fbee295)) but that approach was abandoned eventually as the cause of crashes then was falsely attributed to `consensus_params` being uninitialized (that didn't turn out to be the case)

Reply:
This commit https://github.com/dashpay/dash/commit/f4013af8636636c6bb30b43d913283e9dcee0134 look over-complex and will bring many pains in future backports.

Please, consider replacement these 2 commits https://github.com/dashpay/dash/pull/5501/commits/eb414ad1eba17a5d667404977219e1eaa5705543 and https://github.com/dashpay/dash/commit/f4013af8636636c6bb30b43d913283e9dcee0134 to this simpler and shorted changes: https://github.com/knst/dash/commit/1c71b089e30f18e0783d2630ed9bce45f31adf78

Consider also this refactoring: https://github.com/knst/dash/commit/0bfefaf893c6053d98ed4fc37c8eb0bb179d3245  to avoid confusion for future changes like this PR.